core/src/retransmit_stage.rs

```diff
@@ -131,45 +132,37 @@ impl RetransmitStats {
     }
 }
 
-// Map of shred (slot, index, type) => list of hash values seen for that key.
-type ShredFilter = LruCache<ShredId, Vec<u64>>;
+struct ShredDeduper<const K: usize> {
+    deduper: Deduper<K, [u8]>,
+    filter: Deduper<K, [u64]>,
```

Comment:
can we get a self-descriptive name here?  i'm going to have to recover what it's "filtering" from usage

Reply:
Added inline comment indicating that this is deduping on:
```
(ShredId, /*0..MAX_DUPLICATE_COUNT:*/ usize)
```
If that is not good enough, suggestions for better name?

Reply:
`version_filter`?  not sure we currently have a term to differentiate duplicate blocks :thinking:

Reply:
or `shred_id_filter`? since that's the data it's operating on.

Reply:
`version_filter` is misleading because shred-version is something else.
renamed to `shred_id_filter`.